From 5cf3622c975a4bf013030029711fb2cc7a85dd8f Mon Sep 17 00:00:00 2001
From: Thomas Tai <thomas.tai@windriver.com>
Date: Fri, 18 Feb 2011 15:36:44 -0500
Subject: [PATCH 2/7] cert-hyp: match vb_config structure with certifiable hypervisor

certifiable hypervisor has difference vb_config structure,
changes is needed to match the structure of certifiable hypervisor.
The fields is select/deselect via kernel options.

Signed-off-by: Thomas Tai <thomas.tai@windriver.com>
---
 include/vbi/interface.h |   30 ++++++++++++++++++++++++++++++
 1 files changed, 30 insertions(+), 0 deletions(-)

diff --git a/include/vbi/interface.h b/include/vbi/interface.h
index 4923d8e..671116f 100644
--- a/include/vbi/interface.h
+++ b/include/vbi/interface.h
@@ -538,6 +538,21 @@ struct config_page_map
 	uint32_t accessPriv;
 };
 
+#if defined(CONFIG_WRHV_CERT)
+/* order in this structure must match the ones in board_q_port_t */
+struct vbPortConfig{
+	uint32_t vbPortId;       /* unique Id starting at 0 */
+	char name[VB_NAMELEN];   /* port name */
+	uint32_t vbId;           /* vb Id where port reside */
+	uint32_t channelId;      /* channel id to which the port is connected */
+	uint32_t direction;      /* SOURCE or DESTINATION */
+	uint32_t protocol;       /* SENDER_BLOCK or RECEIVER_DISCARD */
+	uint32_t maxMsgSize;     /* max messages size in bytes */
+	uint32_t maxNumMsgs;     /* queue length */
+};
+#endif
+
+/* vb_config matched with hypervisor include/vbInterface.h */
 struct vb_config
 {
 
@@ -627,6 +642,16 @@ struct vb_config
 	/* number of interrupts connected to this board */
 	uint32_t num_ints;
 
+#if defined(CONFIG_WRHV_CERT)
+	/*
+	 * ports configuration
+	 *--------------------
+	 */
+
+	/* number of ports configured on this board */
+	uint32_t numPorts;
+#endif
+
 	/*
 	 *  misc settings
 	 */
@@ -655,9 +680,14 @@ struct vb_config
 
 	uint32_t numDevices;
 
+#if defined(CONFIG_WRHV_CERT)
+	/* ports configuration information */
+	VB_ALIGN_FIELD_64(struct vbPortConfig *portConfig, pad12);
+#else
 	/* provide mappings for all configuration pages */
 	struct config_page_map configPageMap[MAX_VB_CONFIG_REGIONS];
 	uint32_t configPageNum;
+#endif
 
 };
 
-- 
1.6.5.2

